### PR TITLE
Fix warnings from clang.

### PR DESCRIPTION
**Pull request by Rombur**
Fix warnings from jenkins.

### CODE REVIEW

File: packages/Search/src/DTK_LinearBVH_decl.hpp [hunk 1]
```diff
@@ -33,7 +33,7 @@ struct TreeTraversal;
  * Bounding Volume Hierarchy.
  */
 template <typename DeviceType>
-struct BVH
+class BVH
```
Comment (dalg24) [inline packages/Search/src/DTK_LinearBVH_decl.hpp]: Was about this change?

Comment (Rombur) [inline packages/Search/src/DTK_LinearBVH_decl.hpp]: clang complains that the explicit instantiation is for a class not for a struct

Comment (dalg24) [inline packages/Search/src/DTK_LinearBVH_decl.hpp]: OK, so in order to be consistent with the explicit instantiation in the _def file ([here](https://github.com/Rombur/DataTransferKit/blob/d77f255a1be53d29c919dddd48865484c94841ad/packages/Search/src/DTK_LinearBVH_def.hpp#L98)).

Comment (dalg24) [inline packages/Search/src/DTK_LinearBVH_decl.hpp]: class or struct does not really matter.  Please commit that change in a separate commit.
Say the 1st commit is double brace initialization and this one to be consistent with explicit instantiation.

**Pull request merged**